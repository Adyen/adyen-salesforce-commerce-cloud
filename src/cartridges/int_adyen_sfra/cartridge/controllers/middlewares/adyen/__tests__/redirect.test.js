/* eslint-disable global-require */
let req;
let res;
let redirect;

beforeEach(() => {
  const { adyen } = require('../../index');
  redirect = adyen.redirect;
  jest.clearAllMocks();
  req = {
    querystring: {
      signature: 'some_mocked_url/signature __ mocked_orderNo',
      merchantReference: 'mocked_merchantReference',
    },
  };

  res = {
    redirect: jest.fn(),
  };

});

afterEach(() => {
  jest.resetModules();
});

describe('Redirect', () => {
  it('should fail if there is no order and signature', () => {
    const OrderMgr = require('dw/order/OrderMgr');
    const Logger = require('dw/system/Logger');
    OrderMgr.getOrder.mockImplementation(() => null);
    redirect(req, res, jest.fn());
    expect(Logger.error.mock.calls).toMatchSnapshot();
  });
  it('should fail if signature doesnt match', () => {
    const Logger = require('dw/system/Logger');
    req.querystring.signature = 'mocked_wrong_signature';
    redirect(req, res, jest.fn());
    expect(Logger.error.mock.calls).toMatchSnapshot();
  });
  it('should redirect on valid signature', () => {
    const Logger = require('dw/system/Logger');
    redirect(req, res, jest.fn());
    expect(res.redirect).toBeCalledWith('https://some_mocked_url/signature');
    expect(Logger.error).toBeCalledTimes(0);
  });
});
