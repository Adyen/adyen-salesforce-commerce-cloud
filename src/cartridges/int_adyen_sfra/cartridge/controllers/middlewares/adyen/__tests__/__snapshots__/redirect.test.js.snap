// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirect should fail if signature doesnt match 1`] = `
Array [
  Array [
    "Redirect signature is not correct",
  ],
]
`;

exports[`Redirect should fail if there is no order and signature 1`] = `
Array [
  Array [
    "No signature or no order with orderNo mocked_merchantReference",
  ],
  Array [
    "Redirect signature is not correct",
  ],
]
`;
